### PR TITLE
Beta 0.6.3.8

### PR DESCRIPTION
**Pull request by iajrz**
Includes the in-memory cache implementation to reduce CPU resource usage.

### CODE REVIEW

File: types/config.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ type PocketConfig struct {
 	ABCILogging              bool   `json:"abci_logging"`
 	RelayErrors              bool   `json:"show_relay_errors"`
 	DisableTxEvents          bool   `json:"disable_tx_events"`
+	Cache                    bool
```
Comment (oten91) [inline types/config.go]: Missing metadata for json `json:"cache"`, if not present is presented as  "Cache": false on the config.json.

File: app/config.go [hunk 2]
```diff
@@ -71,9 +71,10 @@ const (
 	DefaultGenesisType
 )
 
-func InitApp(datadir, tmNode, persistentPeers, seeds, remoteCLIURL string, keybase bool, genesisType GenesisType) *node.Node {
+func InitApp(datadir, tmNode, persistentPeers, seeds, remoteCLIURL string, keybase bool, genesisType GenesisType, useCache bool) *node.Node {
 	// init config
 	InitConfig(datadir, tmNode, persistentPeers, seeds, remoteCLIURL)
+	GlobalConfig.PocketConfig.Cache = useCache
```
Comment (oten91) [inline app/config.go:77]: Move this to InitConfig to keep it working like other flags that override, currently only allows to toggle cache with the flag but is present on the config.

**Review by oten91 — CHANGES_REQUESTED** [review]
Also please check CI some tests are failing with invalid or nil pointer references

This could be due to the test setup but let's double-check
```

--- FAIL: TestQueryRelay (3.10s)
    --- FAIL: TestQueryRelay/query_relay_amino (3.10s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xcdd793]
```

**Review by oten91 — APPROVED** [review]
